src/lib.rs

```diff
@@ -1994,3 +2004,126 @@ pub fn compound_reflection_coefficient(ion: &PyDict, targets: Vec<&PyDict>, targ
+    let Z2_ = Z2.clone(); 
+    let number_target_species = Z2.len();
+
+    let options = Options::default_options(true);
+
+    let y = 0.0;
+    let z = 0.0;
+
+    let ux = (angle/180.0*PI).cos() + DELTA;
+    let uy = (angle/180.0*PI).sin() - DELTA;
+    let uz = 0.0;
+
+    let mut direction = Vector::new(ux, uy, uz);
+    direction.normalize();
+
+    let material_parameters = material::MaterialParameters {
+        energy_unit: "EV".to_string(),
+        mass_unit: "AMU".to_string(),
+        Eb: Eb2,
+        Es: Es2,
+        Ec: Ec2,
+        Ed: vec![0.0; number_target_species],
+        Z: Z2,
+        m: m2,
+        interaction_index: vec![0; number_target_species],
+        surface_binding_model: SurfaceBindingModel::AVERAGE,
+        bulk_binding_model: BulkBindingModel::INDIVIDUAL,
+    };
+
+    let geometry_input = geometry::Mesh0DInput {
+        length_unit: "M".to_string(),
+        densities: target_number_densities,
+        electronic_stopping_correction_factor: 1.0
+    };
+
+    let m = material::Material::<Mesh0D>::new(&material_parameters, &geometry_input);
+
+    let x = -m.geometry.energy_barrier_thickness;
+
+    let num_reflected = Mutex::new(0);
+    let energy_reflected = Mutex::new(0.0);
+    let num_sputtered: Mutex<Vec<f64>> = Mutex::new(vec![0.0; targets.len()]);
+
+
+    (0..num_samples as u64).into_par_iter().for_each( |index| {
+
+        let p = particle::Particle::default_incident(
+            m1,
+            Z1,
+            energy,
+            Ec1,
+            Es1,
+            x,
+            ux,
+            uy,
+            uz
+        );
+
+        let output = bca::single_ion_bca(p, &m, &options);
+
+        for particle in output {
+            //println!("output");
+            if particle.E > 0.0 && particle.dir.x < 0.0 && particle.left {
+                if (particle.incident) {
+                    //println!("reflected");
+                    let mut num_reflected = num_reflected.lock().unwrap();
+                    *num_reflected += 1;
+                    let mut energy_reflected = energy_reflected.lock().unwrap();
+                    *energy_reflected += particle.E;
+                }
+                else{
+                    //println!("sputtered");
+                    let mut num_sputtered = num_sputtered.lock().unwrap();
+                    let ind = find_index(particle.Z, &Z2_).unwrap();
+
+                    num_sputtered[ind] += 1.0;
+                    //println!("num_sputtered: {:?}", *num_sputtered);
+                }
+            }
+        }
+    });
+
+
+
+    let num_reflected = *num_reflected.lock().unwrap();
+    let energy_reflected = *energy_reflected.lock().unwrap();
+    let num_sputtered = num_sputtered.lock().unwrap();
+    let num_sputtered_: Vec<f64> = num_sputtered.iter().map(|&x| x / num_samples as f64).collect();

```

Comment:
I'd recommend a name change of this variable from `num_sputtered_` to `sputtering_yield` or `partial_sputtering_yields` or similar. It's not super important, but since it's a return value, someone looking at the last line of the function might expect total counts instead of yields